packages/react-components/react-portal/src/virtualParent/setVirtualParent.test.ts

```diff
@@ -10,7 +10,7 @@ describe('setVirtualParent', () => {
     setVirtualParent(child, virtualParent);
 
     // Assert
-    expect(child).toMatchObject({ _virtual: { parent: virtualParent } });
+    expect(child).toEqual(expect.objectContaining({ _virtual: { parent: virtualParent } }));
```

Comment:
starting with jest/ts-jest 29 looks like `toMatchObject` has some issues when using with DOM

<img width="614" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/481fadc9-d2de-444e-a593-062c1746589d">

I found only following one which was fixed:
https://github.com/jestjs/jest/issues/6730

Also running into this one:
https://github.com/jestjs/jest/issues/14133

I refactored the code to more appropriate matcher and which also better exposed wrong test for `undefined`